### DISCUSSION TITLE
Need for query and filter options for catalogs

### DISCUSSION THREAD

**Discussion by SebastianOpriel** [International-Data-Spaces-Association/ids-specification#74] [Ideas]
I would highly recommend to implement capabilities to limit the returned number of catalog entries and also to provide query functionalities. We have heared several times, that customers want to offer thausands of assets. Each catalog request will lead to a huge response body. If a consumer is querying all connectors available in a dataspace (to gain an overview of offerings), a ones own catalog could simply be exlpoding by the sheer amount of offerings, just provided by one provider. 
This is not a problem, which just we face. I expect that this will be a crucial functionality for almost every data space like C-X, MDS and others. And yes, your answer might be "If you need this functionality, you could implement that on your own" - we would need to implement such functionality for our customers.

https://github.com/International-Data-Spaces-Association/ids-specification/blob/e8b4a9e6088c5f7847dc47ff2f42ecaa2b0df932/catalog/catalog.protocol.md?plain=1#L155-L157

**Comment by SebastianOpriel** [comment]
Nevertheless there is a `dspace:filter` option: https://docs.internationaldataspaces.org/dataspace-protocol/catalog/catalog.binding.https#2.3.1-post

And paginations are also provided, which is pretty good for the doubt described above. https://docs.internationaldataspaces.org/dataspace-protocol/catalog/catalog.binding.https#3.3-pagination

**Comment by matgnt** [comment]
Not the exact answer to your question, but the thread contains some relevant discussion:
#10
